### PR TITLE
Check the length of the data when deserializing

### PR DESCRIPTION
**Pull request by roy-cohen273**

### CODE REVIEW

File: serialization.py [hunk 1]
```diff
@@ -21,9 +28,7 @@ def serialize_bytes(buf: bytearray, data: bytes):
 def deserialize_bytes(buf: bytearray) -> bytes:
     """Deserialize a bytes object and remove it from the buffer."""
     i = deserialize_int(buf)
-    result = buf[:i]
-    del buf[:i]
-    return bytes(result)
+    return bytes(take(buf, i))
```
[the remark this answers is not in the document]
Comment (roy-cohen273) [reply serialization.py:31]: `deserialize_string` checks that the resulting data is a valid string by calling `bytes.decode()`. If non-valid bytes are received, a `UnicodeDecodeError` will be raised with the index and value of the offending byte. I could catch this error and throw my own error, but I think the error thrown by `bytes.decode()` is fitting.

File: serialization.py [hunk 2]
```diff
@@ -3,14 +3,21 @@
 import struct
 
 
+def take(buf: bytearray, length: int) -> bytearray:
```
[the remark this answers is not in the document]
Comment (roy-cohen273) [reply serialization.py]: https://github.com/roy-cohen273/cardazim/pull/2/commits/085f84bbfd98e3d4031f32d9935fb6302f83c88a

File: serialization.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cardazim) [reply serialization.py]: please consider this example, and validate your reasoning:

```
b = bytearray()
st = "hello"
i = 1
serialize_int(b, i)
serialize_string(b, st)
print(deserialize_string(b))
```

[the remark this answers is not in the document]
Comment (roy-cohen273) [reply serialization.py]: `'\x05'` is valid UTF8 (as far as I know), so it makes sense for `deserialize_string` to return it in this example. However, it is not a printable character, and that could be a sign that something went wrong.
I'll add checks for printability in `serialize_string` and `deserialize_string`.

(Note that in this example, the error could be found with `assert_deserialization_finished`.)

**Review by Cardazim — APPROVED** [review]

**Pull request merged**